src/content/app/genome-browser/components/browser-sidebar-modal/modal-views/navigate-modal/GenomeBrowserNavigationButtons.tsx

```diff
@@ -0,0 +1,199 @@
+ * limitations under the License.
+ */
+
+import React, { useEffect, useRef } from 'react';
+
+import { useAppSelector } from 'src/store';
+import useGenomeBrowser from 'src/content/app/genome-browser/hooks/useGenomeBrowser';
+import { useGbRegionQuery } from 'src/content/app/genome-browser/state/api/genomeBrowserApiSlice';
+
+import {
+  getBrowserActiveGenomeId,
+  getActualChrLocation
+} from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+
+import BrowserNavButton from 'src/content/app/genome-browser/components/browser-nav-button/BrowserNavButton';
+import BrowserReset from 'src/content/app/genome-browser/components/browser-reset/BrowserReset';
+
+import type { ChrLocation } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
+
+import styles from './GenomeBrowserNavigationButtons.scss';
+
+type BrowserLocation = {
+  regionLength: number;
+  regionName: string;
+  start: number;
+  end: number;
+};
+
+/**
+ * General considerations:
+ * - left/right buttons move the genome browser 10% of the viewport width left or right
+ * - zoom in / zoom out buttons zoom in or out by 30% of the visible sequence length
+ */
+
+const GenomeBrowserNavigationButtons = () => {
+  const activeGenomeId = useAppSelector(getBrowserActiveGenomeId) as string;
+  const browserLocation = useAppSelector(getActualChrLocation) as ChrLocation;
+  const { changeBrowserLocation } = useGenomeBrowser();
+  const userInputInProgressRef = useRef(false);
+  const userInputTimeoutRef = useRef<ReturnType<typeof setTimeout> | null>(
+    null
+  );
+
+  // Chromosome locations in redux are updated by messages from the genome browser.
+  // These may come too slow for a user who is hammering away at a navigation button.
+  // Therefore, for a more consistent user experience, we are creating a local copy of browser location just for this component
+  const browserLocationRef = useRef<BrowserLocation>({
+    regionLength: Infinity,
+    regionName: browserLocation ? browserLocation[0] : '', // just being extra cautious when accessing parts of the browser location
+    start: browserLocation ? browserLocation[1] : 1,
+    end: browserLocation ? browserLocation[2] : 100
+  });
+
+  const { currentData: regionData } = useGbRegionQuery({
+    genomeId: activeGenomeId,
+    regionName: browserLocation[0] ?? ''
+  });
+
+  useEffect(() => {
+    if (
+      (browserLocation &&
+        browserLocation[0] !== browserLocationRef.current.regionName) ||
+      !userInputInProgressRef.current
+    ) {
+      const [regionName, start, end] = browserLocation;
+      browserLocationRef.current = {
+        regionLength: regionData?.region.length ?? Infinity,
+        regionName,
+        start,
+        end
+      };
+    }
+  }, [activeGenomeId, browserLocation, regionData?.region.length]);
+
+  // TODO: add query for the region so as to never exceed the region length
```

Comment:
Is this for the future?

Reply:
No, that's done. Odd; I was sure I had deleted those comments. At least I removed them now ✅